app/src/androidTest/java/ch/epfl/sdp/peakar/points/POICacheTest.java

```diff
@@ -0,0 +1,147 @@
+
+        GeoPoint geoPoint_4 = new GeoPoint(POINTE_DE_LAPAZ_LAT, POINTE_DE_LAPAZ_LONG,POINTE_DE_LAPAZ_ALT);
+        POIPoint point_4 = new POIPoint(geoPoint_4);
+        point_4.setName(POINTE_DE_LAPAZ_NAME);
+
+        inputArrayList = new ArrayList<>();
+        inputArrayList.add(point_2);
+        inputArrayList.add(point_1);
+        inputArrayList.add(point_3);
+        inputArrayList.add(point_4);
+        userPoint.setLocation(
+                MOCK_LOCATION_LAT_CHAMONIX,
+                MOCK_LOCATION_LON_CHAMONIX,
+                MOCK_LOCATION_ALT_CHAMONIX,
+                0);
+        BoundingBox boundingBox = userPoint.computeBoundingBox(GeonamesHandler.DEFAULT_RANGE_IN_KM);
+        //Create mock cache content
+        POICacheContent poiCacheContent = new POICacheContent(inputArrayList,boundingBox);
+        Gson gson = new Gson();
+        //Convert to JSON
+        String serializesCache = gson.toJson(poiCacheContent);
+        File dataDir = context.getCacheDir();
+        File outputFile = new File(dataDir,CACHE_FILE_NAME_TEST);
+        //Save serialized version
+        try{
+            FileWriter writer = new FileWriter(outputFile);
+            writer.append(serializesCache);
+            writer.flush();
+            writer.close();
+        }
+        catch(IOException e){
+            Log.e("Exception", "File write failed cache POIPoints: " + e.toString());
+        }
+
+    }
+
+
+    @AfterClass
+    public static void cleanup(){
+        //Remove manually created file
+        File cacheDir = ApplicationProvider.getApplicationContext().getCacheDir();
+        File outputFile = new File(cacheDir,CACHE_FILE_NAME_TEST);
+        //noinspection ResultOfMethodCallIgnored
+        outputFile.delete();
+    }
+
+
+    /*Test get instance method*/
+    @Test
+    public void testPOICacheGetInstance(){
+        POICache poiCache = POICache.getInstance();
+        assertNotNull(poiCache);
+    }
+
+    /*Test the saving and the retrieving to/from cache*/
+    @Test
+    public void testPOICacheSaveRetrievePOIDataToCache(){
+        POICache poiCache = POICache.getInstance();
+
+        BoundingBox boundingBox = UserPoint.getInstance(context).computeBoundingBox(GeonamesHandler.DEFAULT_RANGE_IN_KM);
+        File cacheDir = context.getCacheDir();
+        File outputFile = new File(cacheDir,CACHE_FILE_NAME_TEST);
+        //delete old file
+        //noinspection ResultOfMethodCallIgnored
+        outputFile.delete();
+        //Create new cache file
+        poiCache.savePOIDataToCache(inputArrayList,boundingBox,context);
+        ArrayList<POIPoint> result = poiCache.getCachedPOIPoints(context);
+
+        assertEquals(inputArrayList.size(), result.size());
+        assertTrue(inputArrayList.contains(result.get(0)));
+        assertTrue(inputArrayList.contains(result.get(1)));
+        assertTrue(inputArrayList.contains(result.get(2)));
+        assertTrue(inputArrayList.contains(result.get(3)));
+    }
```

Comment:
I am a bit petty. But when I think about this test. Say that both the delete function and the add function doesn't work. So you never remove the item. (and also never create it) then the test still passes right? 
That should be fixed. Maybe check array list before adding the item again. Or splitting it to several tests.

Reply:
I think I will rework the tests... some are still problematic